### PR TITLE
Application form skeleton

### PR DESCRIPTION
**Pull request by auramo**
Shows text fields and sections from database

### CODE REVIEW

File: src/clj/ataru/hakija/hakija_routes.clj [hunk 1]
```diff
@@ -1,12 +1,21 @@
...
-            [ring.util.http-response :refer [ok]]
+            [ring.util.http-response :refer [ok not-found]]
             [compojure.route :as route]
             [selmer.parser :as selmer]))
 
 (def ^:private cache-fingerprint (System/currentTimeMillis))
 
+(defn- fetch-form [id]
+  (let [form (form-store/fetch-form id)]
+    (println "form on server" form)
```
Comment (starvinraven) [inline src/clj/ataru/hakija/hakija_routes.clj]: Remove logging?

File: src/clj/ataru/hakija/hakija_routes.clj [hunk 2]
```diff
@@ -17,15 +26,16 @@
                :tags [{:name "application-api" :description "Application handling"}]}}
     (api/context "/api" []
                  :tags ["application-api"]
-                 (api/GET "/form/:id" [id]
-                          (ok
-                            {:form {:x id}})))))
+                 (api/GET "/form/:id" []
+                          :path-params [id :- Long]
+                          :return s/Any
+                          (ok (fetch-form id))))))
```
Comment (starvinraven) [inline src/clj/ataru/hakija/hakija_routes.clj]: `fetch-form` seems to return a `ok` or `not-found` response, so is the wrapping needed here?

File: src/clj/ataru/hakija/hakija_routes.clj (hunk 1 above)
Comment (auramo) [inline src/clj/ataru/hakija/hakija_routes.clj]: done

File: src/clj/ataru/hakija/hakija_routes.clj (hunk 2 above)
Comment (auramo) [inline src/clj/ataru/hakija/hakija_routes.clj]: Good! Removing...

File: src/cljs/ataru/hakija/application_handlers.cljs [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    (str "/hakemus/api/form/" form-id)
+    :application/handle-form)
+  db)
+
+(register-handler
+  :application/get-form
+  get-form)
+
+(defn handle-form [db [_ form-response]]
+  (println "response when not found" form-response)
```
Comment (starvinraven) [inline src/cljs/ataru/hakija/application_handlers.cljs]: Logging again?

File: src/cljs/ataru/hakija/form_view.cljs [hunk 4]
```diff
@@ -1,9 +1,44 @@
...
+
+(defn render-field
+  [content]
+   (match [content]
+          [{:fieldClass "wrapperElement"
+            :children   children}] [wrapper-field content children]
+          [{:fieldClass "formField" :fieldType "textField"}] [text-field content]))
+
+(defn render-fields [form-data]
+  (if form-data
```
Comment (starvinraven) [inline src/cljs/ataru/hakija/form_view.cljs]: Could just use `when` here?

**Comment by starvinraven** [comment]
Looks good to me! :+1:

File: src/cljs/ataru/hakija/form_view.cljs (hunk 4 above)
Comment (auramo) [inline src/cljs/ataru/hakija/form_view.cljs]: Done

File: src/cljs/ataru/hakija/application_handlers.cljs (hunk 3 above)
Comment (auramo) [inline src/cljs/ataru/hakija/application_handlers.cljs]: removed

**Pull request closed without merging**